.github/workflows/python-validate.yml

```diff
@@ -77,3 +77,63 @@ jobs:
 
           python3 -m pip install pytest
           python3 -m pytest
+
+      - name: Tag if new version
```

Comment:
Tagging should happen only in _integrate_ builds that are run on all commits to `main` (we should not attempt taggin on PRs).

Also we should tag only when all tests (including integration if there are any) pass (see e.g. https://github.com/serverless/console/blob/4ba7158b9e0db8bfd06db031eb30af76cea8de5d/.github/workflows/node-integrate.yml#L514 and `needs` section)